### DISCUSSION TITLE
Include RadioLib in Microchip Studio Project

### DISCUSSION THREAD

**Discussion by pete20211207** [jgromes/RadioLib#493] [Q&A]
Hi Jan,
what is the easiest way to include RadioLib in a Microchip Studio (Atmel Studio) Project?
Background:
The prototype with an Arduino Mega2560 works well - hence on resources issues, I need to continue with an ATmega1281.
So I also need to change from Arduino IDE to Microchip Studio.
How can I use there your valueable library?
Thank you in advance
Peter

**Comment by jgromes** [comment]
Unfortunately I don't think it's going to be that easy - I don't think Microchip solution will support ATmega1281 as an Arduino out-of-the box. RadioLib expects Arduino-style hardware interface.

There's two way to go about this:

1. Writing a custom platform in RadioLib. There's a section in BuildOpt.h that deals with this. You would have to write your own "Arduino-like" layer using the hardware abstraction that Microchip uses. So for example, you would create your own `pinMode` function in the global scope and then provide it's signutare (return type, function name and all arguments) to the `RADIOLIB_CB_ARGS_PIN_MODE` macro - and so on for everything defined below:

https://github.com/jgromes/RadioLib/blob/9117c6125474db6b29f13c7874e6494377661005/src/BuildOpt.h#L42-L79

2. You could use https://github.com/MCUdude/MegaCore, which supports ATmega1281. However, RadioLib doesn't support MegaCore at the moment, only MegaCoreX. Seems like it should be easy to add that support though, just another section in BuildOpt.

**Reply by jgromes** [reply]
So I just checked MegaCore and it compiles fine with RadioLib, so you should be able to use that.

**Reply by pete20211207** [reply]
Sounds great, Jan!
I will try to use the AVR ISP header to use MegaCore to program the ATmega1281.
Thank you a lot so far!
KR Peter

**Reply by pete20211207** [reply]
MegaCore works great with Arduino IDE 1.8.16 and RadioLib 5.1.2!
I exported the hex-file (Sketch -> Export compiled Binary) and programmed it with Microchip Studio (JTAG Interface) and Atmel-ICE.
Thank you again & Happy Easter!
KR Peter
